cmd/release/main.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+)
+
+/*
+required parameteres
+
+https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line
+github token for modifying release
+
+# dependencies
+* requires a build environment (make artifact-release must work)
+* github.com/itchio/gothub
+
+*/
+
+const (
+	// TODO figure this out based on directory name
+	repository   = "cluster-api-provider-aws"
+	artifactsDir = "out"
+)
+
+func main() {
+	// TODO consider moving all these things out to some config struct so other
+	// providers can reuse this release
+	expectedFiles := []string{
+		// TODO these aren't examples. rename them.
+		"cluster-api-provider-aws-examples.tar",
+		"clusterawsadm-darwin-amd64",
+		"clusterawsadm-linux-amd64",
+		"clusterctl-darwin-amd64",
+		"clusterctl-linux-amd64",
+	}
+
+	// TODO it would be ideal if we could release major/minor/patch and have it
+	// automatically bump the latest tag git finds
+	// until then, hard code it
+	remote := os.Args[1] // default to origin probably
+	version := os.Args[2]
+
+	// Build all the release binaries
+	RunCommand("make", "release-artifacts")
+
+	// Creates git tag
+	RunCommand("git", "tag", "--force", version)
+
+	// Push git tag
+	RunCommand("git", "push", remote, version)
+
+	// create draft release
+	// TODO: not everyone names the remotes after the username
+	RunCommand("gothub", "release", "--tag", version, "--user", remote, "--repo", repository, "--draft")
```

Comment:
Would overriding the upstream for the working branch (using `git branch --set-upstream-to`affect this? For example I override the upstream for my master branch to `upstream/master`?

Reply:
Nope, that would be fine. Since this is a github specific release tool, it will assume the remote as "github.com/<USER>/<REPO>" regardless of local git settings. So I think you're pointing at a misconception that I had when writing this -- it actually *is* GitHub user and my variable name is bad. Will fix.